### PR TITLE
throw bounce change

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Another throw pr, time for 6 months of bugs.

Throwing things into dense turfs or objects now makes them bounce off in a semi realistic way.
Currently, the thrown thing bounces DIRECTLY back at the thrower, which although comical for grenade FF, makes absolutely no sense as you watch things bounce in bizarre and impossible ways.

Or if you prefer quality pictures, we get blue instead of red.
image

ALso some mild reorg of some related code. Everything about throwing is fucking god awful though and need more of a rework but that's for another day.

Additionally fixes some exploits regarding grabbing items before they bounce.
I swear there was a bug report or 2 to close, but I can't find them.
## Why It's Good For The Game
More realistic/less guh throw bounce mechanics
## Changelog
:cl:
fix: Objects will no longer bounce at impossible angles when hitting solid turfs or objects
/:cl:

### CODE REVIEW

**Comment by iforgotmeword** [comment]
Greatest buffs to grenades 
(Also sentinels are going to have a lot of fun with this)

**Comment by ilikesaying** [comment]
Map-wide NoobTube bouncing when

**Comment by blackdav123** [comment]
do huggers still bounce off marines? (https://github.com/tgstation/TerraGov-Marine-Corps/issues/13686)

File: code/game/atoms_movable.dm [hunk 1]
```diff
@@ -463,25 +463,41 @@
...
+	if(!isturf(loc))
+		return
+	var/dir_to_proj = get_dir(hit_atom, old_throw_source)
+	if(ISDIAGONALDIR(dir_to_proj))
+		var/list/cardinals = list(turn(dir_to_proj, 45), turn(dir_to_proj, -45))
+		for(var/direction in cardinals)
+			var/turf/turf_to_check = get_step(hit_atom, direction)
+			if(turf_to_check.density)
+				cardinals -= direction
+		dir_to_proj = pick(cardinals)
```
Comment (QualityVan) [inline code/game/atoms_movable.dm:487]: This is gonna runtime if both directions are somehow occupied. I don't know if that's actually possible with how diagonal movement works, mind you.

Comment (Lumipharon) [reply code/game/atoms_movable.dm:487]: yeah it shouldn't be possible, otherwise you wouldn't hit the thing in question to begin with.

**Pull request merged**